catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/EntityRepository.java

```diff
@@ -551,6 +553,15 @@ private void updateDescription() throws JsonProcessingException {
       recordChange("description", original.getDescription(), updated.getDescription());
     }
 
+    private void updateDeleted() throws JsonProcessingException {
+      if (original.getDeleted()) {
+        majorVersionChange = true;
```

Comment:
Does the specific entity repository initialize a new EntityUpdater for every update?
If yes, this logic is okay.

Reply:
Hi @mithmatt, yes! At each `PUT` call we end up in the `createOrUpdate` method in the EntityRepository, where a new updater is created based on the incoming and original entities: https://github.com/open-metadata/OpenMetadata/blob/main/catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/EntityRepository.java#L309